backend/src/main/java/org/ukma/spring/crooodle/service/impl/RoomServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+package org.ukma.spring.crooodle.service.impl;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.ukma.spring.crooodle.dto.*;
+import org.ukma.spring.crooodle.dto.common.PageResponseDto;
+import org.ukma.spring.crooodle.entities.HotelEntity;
+import org.ukma.spring.crooodle.entities.RoomEntity;
+import org.ukma.spring.crooodle.repository.HotelRepository;
+import org.ukma.spring.crooodle.repository.RoomRepository;
+import org.ukma.spring.crooodle.service.RoomService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@Service
+public class RoomServiceImpl implements RoomService {
+
+    private final RoomRepository roomRepository;
+    private final HotelRepository hotelRepository;
+
+    @Override
+    public int create(Long hotelId, RoomCreateRequestDto requestDto) {
+        HotelEntity hotel = hotelRepository.findById(hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Hotel not found"));
+
+        RoomEntity room = RoomEntity.builder()
+            .name(requestDto.getName())
+            .capacity(requestDto.getCapacity())
+            .pricePerNight(requestDto.getPricePerNight())
+            .description(requestDto.getDescription())
+            .available(requestDto.isAvailable())
+            .hotel(hotel)
+            .build();
+
+        room = roomRepository.save(room);
+        return room.getId().intValue();
+    }
+
+    @Override
+    public RoomAdminResponseDto getAdmin(Long hotelId, Long roomId) {
+        RoomEntity room = roomRepository.findByIdAndHotelId(roomId, hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Room not found in the given hotel"));
+        return mapToAdminResponseDto(room);
+    }
+
+    @Override
+    public void edit(Long hotelId, Long roomId, RoomEditRequestDto requestDto) {
+        RoomEntity room = roomRepository.findByIdAndHotelId(roomId, hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Room not found in the given hotel"));
+
+        room.setName(requestDto.getName());
+        room.setCapacity(requestDto.getCapacity());
+        room.setPricePerNight(requestDto.getPricePerNight());
+        room.setDescription(requestDto.getDescription());
+        room.setAvailable(requestDto.isAvailable());
+
+        roomRepository.save(room);
+    }
+
+    @Override
+    public void delete(Long hotelId, Long roomId) {
+        RoomEntity room = roomRepository.findByIdAndHotelId(roomId, hotelId)
+            .orElseThrow(() -> new IllegalArgumentException("Room not found in the given hotel"));
+
+        roomRepository.delete(room);
+    }
+
+    @Override
+    public PageResponseDto<RoomAdminResponseDto> listAdmin(Long hotelId, RoomCriteriaDto criteriaDto) {
+        List<RoomEntity> rooms = roomRepository.findByHotelId(hotelId);
```

Comment:
Does not use criteria in any way? Is this your intended behaviour? Than at least put TODO